libs/wingsdk/src/simulator/simulator.ts

```diff
@@ -314,6 +323,36 @@ export class Simulator {
     }
   }
 
+  private async acquireStateLock() {
+    if (this.lockfile) {
+      return; // already locked
+    }
+
+    await mkdir(this.statedir, { recursive: true });
+    try {
+      this.lockfile = await openFileHandle(this.lockfileFilename, "wx");
```

Comment:
I wonder what we can do to make the mechanism more robust to crashes (maybe storing a process ID or timestamp inside that can be read? not sure)

Reply:
I've been looking out there and found https://www.npmjs.com/package/proper-lockfile. They check for staleness by constantly updating the lockfile, and use `mkdir` instead of `open` to make it network-compatible.

Not sure how a process ID in the lockfile would help, though.

I think we are good if we just implement the staleness check. No need for network compatibility, IMO. What do you think?

Reply:
Oh I see, so basically checking if it's stale based on its timestamp. That sounds good to me 👍